components/script/dom/globalscope.rs

```diff
@@ -2227,6 +2231,66 @@ impl GlobalScope {
         }))
     }
 
+    pub fn create_image_bitmap(
+        &self,
+        image: ImageBitmapSource,
+        options: &ImageBitmapOptions,
+    ) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+        if options.resizeWidth.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        if options.resizeHeight.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        let image_bitmap = ImageBitmap::new(&global, 0, 0).unwrap();
```

Comment:
We're creating a 0x0 image bitmap here. We should wait to create the ImageBitmap until we know the size we need to create.

Reply:
We've set it to the canvas image's width and height. The image bitmap now gets created after the options.resize tests for width and height.